### PR TITLE
[DRUP-732] RBAC on API Docs

### LINKED ISSUE #19: Access control for API Docs
The API Catalog should have access control to control who can see each API Doc.  The Apigee Edge API Product should be an attribute of each API Doc so that an administrator can decide to hide API Docs for certain API Products.

### PR DESCRIPTION
**Pull request by Jaesin**
Fixes #19

### CODE REVIEW

File: src/Entity/Form/ApiDocSettingsForm.php
Comment (cnovak) [inline src/Entity/Form/ApiDocSettingsForm.php]:
```suggestion
      '#description' => $this->t("Select how access to an API doc should be limited based on it's associated API product field from the following options:<br />
```

File: src/Entity/Form/ApiDocSettingsForm.php
Comment (cnovak) [inline src/Entity/Form/ApiDocSettingsForm.php]:
```suggestion
        <strong>Never:</strong> Access will not be limited by the API product field. Only the API doc access permissions are used to determine if a user will have access.<br />
```

File: src/Entity/Form/ApiDocSettingsForm.php
Comment (cnovak) [inline src/Entity/Form/ApiDocSettingsForm.php]:
```suggestion
        <strong>Always:</strong> Access to view an API doc will follow the same settings as the API product access control settings.<br />
```

File: src/Entity/Form/ApiDocSettingsForm.php
Comment (cnovak) [inline src/Entity/Form/ApiDocSettingsForm.php]:
```suggestion
        <strong>Configurable:</strong>Access to view an API doc will follow the same settings as the API product access control settings only if the "Restrict access based on API product" is checked on that individual API doc.
```

File: src/Entity/Form/ApiDocSettingsForm.php
Comment (cnovak) [inline src/Entity/Form/ApiDocSettingsForm.php:110]: Made some suggestions to try to make the wording more clear.  Also, would be good to have a link over to the API product access settings page so that the user can understand what we are describing.

File: src/Entity/ApiDoc.php
Comment (cnovak) [inline src/Entity/ApiDoc.php]:
```suggestion
    // Store whether product access should be checked per entity.
```

File: src/Entity/ApiDoc.php
Comment (cnovak) [inline src/Entity/ApiDoc.php:341]:
```suggestion
    // Gets whether the access control should be configurable per entity.
```

File: src/Entity/ApiDoc.php
Comment (cnovak) [inline src/Entity/ApiDoc.php]: Should add a link here also to the API product access settings so that user understands where the API product access control settings are:
```suggestion
      ->setDescription(t('If checked, access to view this API doc will restricted using the API product access control settings.'))
```

File: src/Entity/ApiDoc.php
Comment (cnovak) [inline src/Entity/ApiDoc.php:344]: Should this field be renamed `is_product_access_controlled`?  Not sure if it is standard to follow `getXXXX()` or `isXXXX()` convention for fields.

File: src/Entity/ApiDoc.php
Comment (cnovak) [inline src/Entity/ApiDoc.php:310]: Without the `setDisplayOptions` this field was disabled in the form when enabling the module, which would make it confusing for users to understand when viewing the API product access control setting.

**Review by cnovak — CHANGES_REQUESTED** [review]
Mostly wording changes, but would be good to add links over to API product access control settings tab if possible.

Also, need to add an issue in to take care of the fact that the API Catalog view is displaying APIs even if the access control does not give user access to view:

Steps to repro:
1) Set access control to "Always"
2) Create an API Doc
3) Set permissions for anon users to view published API docs
4) As anonymous user, go to /apis

*Expected:* The /apis list is empty since user is not given access to API Doc
*Actual:* The /apis list shows doc, but when clicking on doc to go to /api/1, access is denied
So, anon user can  view API Doc name/description, which they should not see if the API Doc is an internal API.

File: src/Entity/ApiDoc.php
Comment (Jaesin) [reply src/Entity/ApiDoc.php:310]: OK

File: src/Entity/ApiDoc.php
Comment (Jaesin) [reply src/Entity/ApiDoc.php:344]: There isn't really a precedent for that. This ends up being a DB table name so brevity is usually preferred. For instance, `is_published` is simply stored as `status`.

File: src/Entity/Form/ApiDocSettingsForm.php
Comment (Jaesin) [reply src/Entity/Form/ApiDocSettingsForm.php:110]: Good idea, I added a link to the API product access configuration page.

**Comment by Jaesin** [comment]
I re-based but some of the tests are going to fail. The problem is, we don't have a way to mock API responses for the edit forms since they will try to get a list for API Products. That wasn't a problem when using the autocomplete widget but select widget has to be pre-populated.

**Comment by mxr576** [comment]
Well, the possibility of complex-mocking is waiting for approval in this PR: https://github.com/apigee/apigee-edge-drupal/pull/186

**Comment by arunz6161** [comment]
Need to revisit this after finalizing on testing strategy

**Comment by cnovak** [comment]
This will be fixed by #71 instead, closing this PR.

**Pull request closed without merging**
